tests/test_bulk.py

```diff
@@ -13,6 +13,21 @@ async def test_bulk_create(self):
         inc = all_[0]["id"]
         self.assertEqual(all_, [{"id": val + inc, "name": None} for val in range(1000)])
 
+    async def test_bulk_create_with_specified(self):
+        await UniqueName.bulk_create([UniqueName(id=id_) for id_ in range(1000)])
+        all_ = await UniqueName.all().values("id", "name")
+        self.assertEqual(all_, [{"id": id_, "name": None} for id_ in range(1000)])
+
+    async def test_bulk_create_mix_specified(self):
+        await UniqueName.bulk_create(
+            [UniqueName(id=id_) for id_ in range(1000)] + [UniqueName() for _ in range(1000)]
+        )
+
+        all_ = await UniqueName.all().values("id", "name")
```

Comment:
Please assert that the size of all_ is 2000
Just to ensure that ID-overlap isn't happening?

Reply:
As you said, I think Changing insertion sequence will prevent ID overlapping. But we can beat the stone bridge before you cross. I'll add it soon.